### ISSUE TITLE
New lines in doc string 

### ISSUE THREAD

**Issue by gcarleo** [netket/netket#160]
Hi @ooreilly, we were noticing that new lines are completely removed when parsing the docstring to markdown. For example, a docstring like this: 

```python
>>> print(nk.optimizer.Sgd.__doc__)
Simple Stochastic Gradient Descent Optimizer.
        [Stochastic Gradient Descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)
        is one of the most popular optimizers in machine learning applications.
        Given a stochastic estimate of the gradient of the cost function ($$ G(\mathbf{p}) $$),
        it performs the update:

        $$
        p^\prime_k = p_k -\eta G_k(\mathbf{p}),
        $$

        where $$ \eta $$ is the so-called learning rate.
        NetKet also implements two extensions to the simple SGD,
        the first one is $$ L_2 $$ regularization,
        and the second one is the possibility to set a decay
        factor $$ \gamma \leq 1 $$ for the learning rate, such that
        at iteration $$ n $$ the learning rate is $$ \eta \gamma^n $$.  
```

is rendered in markdown without new lines, and makes the visualization of LateX formulas a bit ugly: 

> Simple Stochastic Gradient Descent Optimizer. [Stochastic Gradient Descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent) is one of the most popular optimizers in machine learning applications. Given a stochastic estimate of the gradient of the cost function ($$ G(\mathbf{p}) $$), it performs the update: $$ p^\prime_k = p_k -\eta G_k(\mathbf{p}), $$ where $$ \eta $$ is the so-called learning rate. NetKet also implements two extensions to the simple SGD, the first one is $$ L_2 $$ regularization, and the second one is the possibility to set a decay factor $$ \gamma \leq 1 $$ for the learning rate, such that at iteration $$ n $$ the learning rate is $$ \eta \gamma^n $$.

Is there a way to have the new lines as they are in the docstring? Thanks!

**Comment by ooreilly** [comment]
Hi @gcarleo, I am not sure what is causing this problem right now. I will need to investigate. Please let me get back to you within the next day with a solution. Thanks.

**Comment by ooreilly** [comment]
@gcarleo I have not yet been able to reproduce the problem. I wrote a test case using your example. I tested extracting the docstring using `extract.py` and then parsing it using `parse.py`. The new line characters are preserved for me.  Maybe it is coming from `format.py`. Unfortunately, I will not be able to check that now.  I will not be able to resume operations until tomorrow evening (timezone: PST). Sorry for the inconvenience.
